PaintTest/main.c

```diff
@@ -0,0 +1,691 @@
+		sprintf(*out, "%s | %d", paletteName, image);
+	}
+}
+
+static void printFunctionCall(utf8string *out, function_call call) {
+	utf8string imageId = malloc(64);
+	printImageId(call.supports.colour_flags, &imageId);
+	switch (call.function) {
+	case SUPPORTS_WOOD_A:
+		sprintf(*out, "wooden_a_supports_paint_setup(%d, %d, %d, %s)", call.supports.type, call.supports.special, call.supports.height, imageId);
+		return;
+	case SUPPORTS_WOOD_B:
+		sprintf(*out, "wooden_a_supports_paint_setup(%d, %d, %d, %s)", call.supports.type, call.supports.special, call.supports.height, imageId);
+		return;
+
+	case SUPPORTS_METAL_A:
+		sprintf(*out, "metal_a_supports_paint_setup(%d, %d, %d, %d, %s)", call.supports.type, call.supports.segment, call.supports.special, call.supports.height, imageId);
+		return;
+	case SUPPORTS_METAL_B:
+		sprintf(*out, "metal_b_supports_paint_setup(%d, %d, %d, %d, %s)", call.supports.type, call.supports.segment, call.supports.special, call.supports.height, imageId);
+		return;
+	}
+
+	utf8string name;
+	switch (call.function) {
+	case PAINT_98196C:
+		name = "sub_98196C";
+		break;
+	case PAINT_98197C:
+		name = "sub_98197C";
+		break;
+	case PAINT_98198C:
+		name = "sub_98198C";
+		break;
+	case PAINT_98199C:
+		name = "sub_98199C";
+		break;
+	}
+
+	int strlen;
+
+	printImageId(call.paint.image_id, &imageId);
+	strlen = sprintf(
+		*out,
+		"%s(%s, %d, %d, %d, %d, %d, %d, ",
+		name,
+		imageId,
+		call.paint.offset.x, call.paint.offset.y,
+		call.paint.bound_box_length.x, call.paint.bound_box_length.y, call.paint.bound_box_length.z,
+		call.paint.z_offset,
+		NULL
+	);
+
+	if (call.function != PAINT_98196C) {
+		strlen += sprintf(
+			*out + strlen,
+			"%d, %d, %d, ",
+			call.paint.bound_box_offset.x, call.paint.bound_box_offset.y, call.paint.bound_box_offset.z
+		);
+	}
+
+	sprintf(*out + strlen, "%d)", call.paint.rotation);
+}
+
+static void printFunctionCallArray(utf8string *out, function_call calls[], uint8 count) {
+	for (int i = 0; i < count; i++) {
+		utf8string callOut = malloc(128);
+		printFunctionCall(&callOut, calls[i]);
+		sprintf(*out + strlen(*out), "%s\n", callOut);
+	}
+}
+
+int getTrackSequenceCount(uint8 rideType, uint8 trackType) {
+	int sequenceCount = 0;
+	const rct_preview_track **trackBlocks;
+
+	if (ride_type_has_flag(rideType, RIDE_TYPE_FLAG_FLAT_RIDE)) {
+		trackBlocks = FlatRideTrackBlocks;
+	} else {
+		trackBlocks = TrackBlocks;
+	}
+
+	for (int i = 0; i < 256; i++) {
+		if (trackBlocks[trackType][i].index == 0xFF) {
+			break;
+		}
+
+		sequenceCount++;
+	}
+
+	return sequenceCount;
+}
+
+bool testTrackElement(uint8 rideType, uint8 trackType, utf8string *error) {
+	if (rideType == RIDE_TYPE_CHAIRLIFT) {
+		if (trackType == TRACK_ELEM_BEGIN_STATION || trackType == TRACK_ELEM_MIDDLE_STATION || trackType == TRACK_ELEM_END_STATION) {
+			// These rides chechk neighbouring tiles for tracks
+			*error = "Skipped";
+			return false;
+		}
+	}
+	uint8 rideIndex = 0;
+	rct_map_element mapElement = {};
```

Comment:
`= {}` should be `= { 0 }` if you want to zero it out and be compatible with VS.